### ISSUE TITLE
Enhancement - Limit the number of entries for a style

### ISSUE THREAD

**Issue by Tot31** [geoffhumphrey/brewcompetitiononlineentry#908]
Hi all,

I'm currently translating the website into french and have one of my own setup for a very local contest.
We will have 5 styles available and 10 entries max per style.

I was wondering if it is possible to set this 10 entries per style boundary somewhere ?
I managed to set the limit for the whole contest to 50 entries and that no more than 5 entries can be added by a user, but can I set a number of maximum entries per style ?

**Comment by geoffhumphrey** [comment]
Yes, there is. Go to Admin > Preferences > Site Preferences. In the Entries section, you can set the maximum entries per sub-style there.

**Comment by buergermeister75** [comment]
I'm going to second this request Geoff. There are some professional comps that want to limit the number of entries per a category now. Say only want to allow 150 for M1, M2, M3, and M4 each.

**Comment by r3dunlop** [comment]
I'm going to third this request. It would be very helpful to put caps on either table, style or substyle. In medium sized competitions with multiple brewery pro-am co-sponsors, you want to make sure that the lager brewery doesn't get 5 entries while the IPA brewery 40, for example.

**Comment by gradys** [comment]
Hi,
I'd highly appreciate this functionality as well.

**Comment by AdrienM76** [comment]
Two years running we've come very close to having a difficult go at planning flights and tables because some categories were entered heavily and others not so much. We got lucky both times with just the right entries happening to withdraw at some point but I'm sure our luck will run out. Having the ability to keep the entries balanced would be welcome.

**Comment by cgspeck** [comment]
I had to implement a category (style) limit function for my brew club, based on the v2.4.0.3 tag you can find the commit [here](https://github.com/cgspeck/brewcompetitiononlineentry/commit/fa6e2a1562e0d629293bf4b5b77d186aabed5c2e). It works by adding a `limit_category` function that checks existing entries for the brewer for the same category, and is callled from `limit_subcategory`. It's only tested against the `AABC` style set and does not handle limit exceptions.

**Comment by rocket-pants** [comment]
Also very keen for a way to set limits per style. We've implemented a 2 entry per style limit (and one per sub-style) for years but have to manually manage the style entry limit. Every year during the state comp registration someone doesn't read the rules and enters more than the limit. Generally it is simple to sort but would be good if there was functionality to apply the limit at time of entry registration
